x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/BasicQueryClient.java

```diff
@@ -49,4 +67,69 @@ public void query(QueryRequest request, ActionListener<Payload> listener) {
         SearchRequest search = prepareRequest(client, searchSource, false, indices);
         client.search(search, new BasicListener(listener));
     }
-}
+
+    @Override
+    public void get(Iterable<List<HitReference>> refs, ActionListener<List<List<SearchHit>>> listener) {
+        MultiGetRequestBuilder requestBuilder = client.prepareMultiGet();
+        // no need for real-time
+        requestBuilder.setRealtime(false)
+                      .setRefresh(false);
+
+        int sz = 0;
+
+        for (List<HitReference> list : refs) {
+            sz = list.size();
+            for (HitReference ref : list) {
+                Item item = new Item(ref.index(), ref.id());
+                // make sure to get the whole source
+                item.fetchSourceContext(FetchSourceContext.FETCH_SOURCE);
+                requestBuilder.add(item);
+            }
+        }
+        
+        final int listSize = sz;
+        client.multiGet(requestBuilder.request(), wrap(r -> {
+            List<List<SearchHit>> hits = new ArrayList<>(r.getResponses().length / listSize);
+            
+            List<SearchHit> sequence = new ArrayList<>(listSize);
+            
+            // copy streams - reused across the whole loop
+            PipedInputStream in = new PipedInputStream();
+            PipedOutputStream out = new PipedOutputStream(in);
+            StreamOutput so = new OutputStreamStreamOutput(out);
+            StreamInput si = new InputStreamStreamInput(in);
+
+            int counter = 0;
+            for (MultiGetItemResponse mgr : r.getResponses()) {
+                if (mgr.isFailed()) {
+                    listener.onFailure(mgr.getFailure().getFailure());
+                    return;
+                }
+                // HACK: the only way to get GetResult is to serialize it and then load it back :(
+                mgr.getResponse().writeTo(so);
+                GetResult result = new GetResult(si);
+
+                SearchHit hit = new SearchHit(-1, result.getId(), result.getDocumentFields(), result.getMetadataFields());
+                hit.sourceRef(result.internalSourceRef());
+                // need to create these objects to set the index
+                hit.shard(new SearchShardTarget(null, new ShardId(result.getIndex(), "", -1), null, null));
+
+                //hit.setSeqNo(result.getSeqNo());
+                //hit.setPrimaryTerm(result.getPrimaryTerm());
+                //hit.version(result.getVersion());
```

Comment:
@jrodewig Normally these need to be set as well however I didn't want to make this PR longer. For event queries we do return these fields however the docs doesn't include them for sequences.
Considering the tight deadline, not sure what's the easiest fix until the next release?

Reply:
Just add a `// TESTRESPONSE[skip: response format changed]` to any failing response for now.

That should skip the CI tests for it. I can go in an fix up any skipped test responses as part of a follow-up PR.

Reply:
I skipped affected docs tests here: 4c3a6bf

Reply:
Thanks @jrodewig. I've removed the comments for said lines - let's see what the build thinks of it.